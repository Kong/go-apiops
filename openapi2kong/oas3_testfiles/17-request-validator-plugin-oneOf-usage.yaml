# When the request-validator is added without a body or parameter schema
# the generator should automatically generate it.

openapi: 3.0.2

info:
  title: Example
  version: 1.0.0

servers:
  - url: http://backend.com/path

x-kong-plugin-request-validator: {}

paths:
  /params/test:
    get:
      x-kong-plugin-request-validator:
        enabled: true
        config:
          body_schema: '{}'
      parameters:
        - in: query
          name: queryid
          schema:
            type: number
            oneOf:
              - type: integer
                example: 10
              - type: number
                example: 2.5
          required: true
        - in: header
          name: testHeader
          schema:
            $ref: '#/components/schemas/headerType'
          required: true
        - in: header
          name: secondTestHeader
          schema:
            $ref: '#/components/schemas/secondHeaderType'
          required: true
components:
  schemas:
    headerType:
      type: string
      oneOf:
        - $ref: '#/components/schemas/stringType'
        - $ref: '#/components/schemas/numberType'
    secondHeaderType:
      $ref: '#/components/schemas/headerType'
    stringType:
      type: string
      example: "10"
    numberType:
      type: number
      example: 2.5
    
