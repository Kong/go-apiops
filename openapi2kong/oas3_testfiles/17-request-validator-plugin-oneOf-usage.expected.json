{
  "_format_version": "3.0",
  "services": [
    {
      "host": "backend.com",
      "id": "730d612d-914b-5fe8-8ead-e6aa654318ef",
      "name": "example",
      "path": "/path",
      "plugins": [],
      "port": 80,
      "protocol": "http",
      "routes": [
        {
          "id": "1446ecde-7037-5f9c-8537-8217e2a12bfa",
          "methods": [
            "GET"
          ],
          "name": "example_params-test_get",
          "paths": [
            "~/params/test$"
          ],
          "plugins": [
            {
              "config": {
                "body_schema": "{}",
                "parameter_schema": [
                  {
                    "explode": true,
                    "in": "query",
                    "name": "queryid",
                    "required": true,
                    "schema": "{\"oneOf\":[{\"example\":10,\"type\":\"integer\"},{\"example\":2.5,\"type\":\"number\"}],\"type\":\"number\"}",
                    "style": "form"
                  },
                  {
                    "explode": false,
                    "in": "header",
                    "name": "testHeader",
                    "required": true,
                    "schema": "{\"$ref\":\"#/definitions/headerType\",\"definitions\":{\"headerType\":{\"oneOf\":[{\"$ref\":\"#/definitions/stringType\"},{\"$ref\":\"#/definitions/numberType\"}],\"type\":\"string\"},\"numberType\":{\"example\":2.5,\"type\":\"number\"},\"stringType\":{\"example\":\"10\",\"type\":\"string\"}}}",
                    "style": "simple"
                  },
                  {
                    "explode": false,
                    "in": "header",
                    "name": "secondTestHeader",
                    "required": true,
                    "schema": "{\"$ref\":\"#/definitions/secondHeaderType\",\"definitions\":{\"numberType\":{\"example\":2.5,\"type\":\"number\"},\"secondHeaderType\":{\"oneOf\":[{\"$ref\":\"#/definitions/stringType\"},{\"$ref\":\"#/definitions/numberType\"}],\"type\":\"string\"},\"stringType\":{\"example\":\"10\",\"type\":\"string\"}}}",
                    "style": "simple"
                  },
                  {
                    "explode": true,
                    "in": "query",
                    "name": "testArrayOne",
                    "required": true,
                    "schema": "{\"items\":{\"oneOf\":[{\"type\":\"string\"},{\"type\":\"integer\"}]},\"type\":\"array\"}",
                    "style": "form"
                  },
                  {
                    "explode": true,
                    "in": "query",
                    "name": "testArrayTwo",
                    "required": true,
                    "schema": "{\"definitions\":{\"headerType\":{\"oneOf\":[{\"$ref\":\"#/definitions/stringType\"},{\"$ref\":\"#/definitions/numberType\"}],\"type\":\"string\"},\"newType\":{\"oneOf\":[{\"type\":\"string\"},{\"type\":\"number\"}],\"type\":\"string\"},\"numberType\":{\"example\":2.5,\"type\":\"number\"},\"stringType\":{\"example\":\"10\",\"type\":\"string\"}},\"items\":{\"oneOf\":[{\"$ref\":\"#/definitions/headerType\"},{\"$ref\":\"#/definitions/newType\"}]},\"type\":\"array\"}",
                    "style": "form"
                  }
                ],
                "version": "draft4"
              },
              "enabled": true,
              "id": "8bd60198-9b34-5f0b-9240-4826c7c331a0",
              "name": "request-validator",
              "tags": [
                "OAS3_import",
                "OAS3file_17-request-validator-plugin-oneOf-usage.yaml"
              ]
            }
          ],
          "regex_priority": 200,
          "strip_path": false,
          "tags": [
            "OAS3_import",
            "OAS3file_17-request-validator-plugin-oneOf-usage.yaml"
          ]
        }
      ],
      "tags": [
        "OAS3_import",
        "OAS3file_17-request-validator-plugin-oneOf-usage.yaml"
      ]
    }
  ],
  "upstreams": []
}